sklearnex/preview/ensemble/forest.py

```diff
@@ -963,6 +1088,14 @@ def _onedal_ready(self, X, y, sample_weight):
             y = np.reshape(y, (-1, 1))
         self.n_outputs_ = y.shape[1]
         ready = self.n_outputs_ == 1
+        if not sp.issparse(X) and sklearn_check_version("1.4"):
+            try:
+                _assert_all_finite(X)
+                ready &= True
+            except ValueError:
+                ready &= False
+        else:
+            ready &= True

```

Comment:
What is the reason to keep this `else` branch?

Reply:
```suggestion
```